src/daft-plan/src/logical_ops/unpivot.rs

```diff
@@ -0,0 +1,94 @@
+use std::sync::Arc;
+
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    datatypes::Field,
+    schema::{Schema, SchemaRef},
+    utils::supertype::try_get_supertype,
+    DataType,
+};
+use daft_dsl::ExprRef;
+
+use itertools::Itertools;
+use snafu::ResultExt;
+
+use crate::{
+    logical_plan::{self, CreationSnafu},
+    LogicalPlan,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Unpivot {
+    pub input: Arc<LogicalPlan>,
+    pub ids: Vec<ExprRef>,
+    pub values: Vec<ExprRef>,
+    pub variable_name: String,
+    pub value_name: String,
+    pub output_schema: SchemaRef,
+}
+
+impl Unpivot {
+    pub(crate) fn try_new(
+        input: Arc<LogicalPlan>,
+        ids: Vec<ExprRef>,
+        values: Vec<ExprRef>,
+        variable_name: &str,
+        value_name: &str,
+    ) -> logical_plan::Result<Self> {
+        if values.is_empty() {
```

Comment:
Why not put the [hashset check logic](https://github.com/Eventual-Inc/Daft/pull/2204/files#diff-c43d64f879714fd53d07744de25a5dd9e9c0c79bddeb03e216b3cba2c9e6ca7eR285-R303) here? then u only need one place to check for values.is_empty

Reply:
Yep this is redundant, I'll remove it here. I'm going to keep the logic for populating `values` in the builder since it requires the schema.